azuredeploy.json

```diff
@@ -0,0 +1,772 @@
+      "artifactsLocationSasToken": "[parameters('_artifactsChinaLocationSasToken')]",
+      "configPrimaryNodeScript": "[uri(parameters('_artifactsChinaLocation'), concat('scripts/config_primary.sh', parameters('_artifactsChinaLocationSasToken')))]",
+      "configSecondaryNodeScript": "[uri(parameters('_artifactsChinaLocation'), concat('scripts/config_secondary.sh', parameters('_artifactsChinaLocationSasToken')))]",
+      "routerNodeScript": "[uri(parameters('_artifactsChinaLocation'), concat('scripts/router.sh', parameters('_artifactsChinaLocationSasToken')))]",
+      "replicaPrimaryNodeScript": "[uri(parameters('_artifactsChinaLocation'), concat('scripts/replica_primary.sh', parameters('_artifactsChinaLocationSasToken')))]",
+      "replicaSecondaryNodeScript": "[uri(parameters('_artifactsChinaLocation'), concat('scripts/replica_secondary.sh', parameters('_artifactsChinaLocationSasToken')))]",
+      "certificateLocation": "[uri(parameters('_artifactsLocation'), concat('scripts/MongoAuthCert.pem', parameters('_artifactsLocationSasToken')))]"
+    },
+    "environment": "[variables(concat('environment', parameters('environment')))]",
+    "templateBaseUrl": "[variables('environment').artifactsLocation]",
+    "sharedTemplateUrl": "[uri(variables('templateBaseUrl'), concat('nested/shared-resources.json', variables('environment').artifactsLocationSasToken))]",
+    "configPrimaryTemplateUrl": "[uri(variables('templateBaseUrl'), concat('nested/config-primary-resources.json', variables('environment').artifactsLocationSasToken))]",
+    "configSecondaryTemplateUrl": "[uri(variables('templateBaseUrl'), concat('nested/config-secondary-resources.json', variables('environment').artifactsLocationSasToken))]",
+    "routerTemplateUrl": "[uri(variables('templateBaseUrl'), concat('nested/router-resources.json', variables('environment').artifactsLocationSasToken))]",
+    "replicaPrimaryTemplateUrl": "[uri(variables('templateBaseUrl'), concat('nested/replica-primary-resources.json', variables('environment').artifactsLocationSasToken))]",
+    "replicaSecondaryTemplateUrl": "[uri(variables('templateBaseUrl'), concat('nested/replica-secondary-resources.json', variables('environment').artifactsLocationSasToken))]",
+    "namespace": "mongodb-",
+    "virtualNetworkName": "myVNET",
+    "networkSettings": {
+      "virtualNetworkName": "[variables('virtualNetworkName')]",
+      "addressPrefix": "10.0.0.0/16",
+      "subnet": {
+        "dse": {
+          "name": "dse",
+          "prefix": "10.0.0.0/24",
+          "vnet": "[variables('virtualNetworkName')]"
+        }
+      },
+      "statics": {
+        "clusterRange": {
+          "base": "10.0.0.",
+          "start": 5
+        },
+        "configPrimaryIp": "10.0.0.240",
+        "configSecondaryIp1": "10.0.0.241",
+        "configSecondaryIp2": "10.0.0.242",
+        "routerIp": "10.0.0.230",
+        "replicaPrimaryIp": "10.0.0.100"
```

Comment:
how are the replica secondary ip's assigned?

Reply:
They are assigned statically as private Ip's to the secondary replica nodes that are used by the primary replica to create a replica set. The private Ip of the primary replica is used by the router server to add a shard